pkg/domain/infra/abi/images_list.go

```diff
@@ -30,9 +31,14 @@ func (ir *ImageEngine) List(ctx context.Context, opts entities.ImageListOptions)
 		if err != nil {
 			return nil, fmt.Errorf("getting repoDigests from image %q: %w", img.ID(), err)
 		}
-		isDangling, err := img.IsDangling(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("checking if image %q is dangling: %w", img.ID(), err)
+
+		isDangling := false
+		if img.ListData.IsDangling != nil {
```

Comment:
Pedantically, I’d prefer `if img.ListData.IsDangling == nil { return … "internal error" }`, not silently filtering images incorrectly.

Or just `isDangling = *img.ListData.IsDangling` without any check, triggering a nil pointer crash. (Alternatively, the c/common-returned data could just use a pointer in the first place, fully trusting the caller to only use valid data.)

Or course, that doesn‘t make a difference in practice, unless there were some other bug in c/common.

Reply:
I like that idea, thanks.  I have to repush in any case since the GitHub failures confused Cirrus.